purchase_requisition_auto_rfq/models/purchase_requisition.py

```diff
@@ -0,0 +1,67 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from collections import defaultdict
+
+from odoo import _, models
+
+
+class PurchaseRequisition(models.Model):
+    _inherit = "purchase.requisition"
+
+    def auto_rfq_from_suppliers(self):
+        """create purchase orders from registered suppliers for products in the
+        requisition.
+
+        The created PO for each supplier will only concern the products for
+        which an existing product.supplierinfo record exist for that product.
+        """
+        self.ensure_one()
+        seller_products, products_without_supplier = self._get_sellers()
+        if products_without_supplier:
+            self._post_products_without_supplier_message(products_without_supplier)
+        return self._create_rfqs(seller_products)
+
+    def _get_sellers(self):
+        seller_products = defaultdict(set)
```

Comment:
Can be done with 
```suggestion
        seller_products = defaultdict(self.env["product.product"])
```
and change below
```python
seller_products[seller.name.id] |= line.product_id
```

But in the end (`_remove_lines_without_official_supplier` function)  doesn't impacts too much and will be the same code.